src/com/mateanna/hw5/Calculator.java

```diff
@@ -0,0 +1,39 @@
+package com.mateanna.hw5;
+
+import jdk.dynalink.Operation;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.BiFunction;
+import java.util.function.BinaryOperator;
+
+public class Calculator {
+
+    private static Map<Character, BinaryOperator<Double>> operations() {
+        BinaryOperator<Double> sum = (x, y) -> x + y;
+        BinaryOperator<Double> multiply = (x, y) -> x * y;
+        BinaryOperator<Double> divide = (x, y) -> x / y;
+        BinaryOperator<Double> substract = (x, y) -> x - y;
+        BinaryOperator<Double> root = (x, y) -> (x / (1 / y));
+        BinaryOperator<Double> power = (x, y) -> (Math.pow(x, (1 / y)));
+        BinaryOperator<Double> power117 = (x, y) -> ((Math.pow(((x + y) / x + 117), y)));
+
+        Map<Character, BinaryOperator<Double>> operatorMap = new HashMap<>();
+        operatorMap.put('+', sum);
+        operatorMap.put('*', multiply);
+        operatorMap.put('/', divide);
+        operatorMap.put('-', substract);
+        operatorMap.put('#', root);
+        operatorMap.put('^', power);
+        operatorMap.put('p', power117);
+
+        return operatorMap;
+    }
+
+    public static double calculate(double x, char operator, double y) {
+        BinaryOperator<Double> op = operations().get(operator);
```

Comment:
write here a condition `if (operations().get(operator) == null)` then throw exception or try to use Optional